cluster-controller/src/main/java/io/strimzi/controller/cluster/operator/resource/StatefulSetOperator.java

```diff
@@ -251,4 +252,38 @@ public String getPodName(StatefulSet desired, int podId) {
         log.debug("Patched {} {}/{}", resourceKind, namespace, name);
         return Future.succeededFuture(ReconcileResult.patched(null));
     }
+
+    /**
+     * Reverts the changes done storage configuration of running cluster. Such changes are not allowed.
+     *
+     * @param current Current StatefulSet
+     * @param desired New StatefulSet
+     */
+    protected void revertStorageChanges(StatefulSet current, StatefulSet desired) {
+        if (current.getSpec().getVolumeClaimTemplates().isEmpty()) {
+            // We are on ephemeral storage and changing to persistent
+            desired.getSpec().setVolumeClaimTemplates(current.getSpec().getVolumeClaimTemplates());
+
+            List<Volume> volumes = current.getSpec().getTemplate().getSpec().getVolumes();
+            for (int i = 0; i < volumes.size(); i++) {
+                Volume vol = volumes.get(i);
+                if ("data".equals(vol.getName()) && vol.getEmptyDir() != null) {
```

Comment:
I know that currently `AstractModel.VOLUME_NAME` is protected and maybe we should make it public if we need it here.

Reply:
Good point. Fixed.